tests/lambda_endpoints/test_batch_starter.py

```diff
@@ -0,0 +1,258 @@
+        cur, "CodiceHi", "imap_l3a_sci_codicehi_20230602_v01.cdf"
+    )
+
+    assert data_level == "l3a"
+    assert version_number == 1
+    assert process_dates == ["2023-05-31", "2023-06-01", "2023-06-02"]
+
+
+def test_all_dependency_present():
+    # Test items in dependencies are all present in result_list.
+    dependencies_true = [
+        {"instrument": "CodiceHi", "level": "l0"},
+        {"instrument": "CodiceHi", "level": "l2"},
+    ]
+    dependencies_false = [
+        {"instrument": "CodiceHi", "level": "l0"},
+        {"instrument": "CodiceHi", "level": "l3"},
+    ]
+
+    result = [
+        {
+            "id": 6,
+            "filename": "imap_l2_sci_codicehi_20230531_v01.cdf",
+            "instrument": "codicehi",
+            "version": 1,
+            "level": "l2",
+            "mode": "NULL",
+            "date": datetime.datetime(
+                2023, 6, 2, 5, 45, tzinfo=zoneinfo.ZoneInfo(key="America/Denver")
+            ),
+            "ingested": datetime.datetime(
+                2023, 6, 2, 5, 45, tzinfo=zoneinfo.ZoneInfo(key="America/Denver")
+            ),
+            "mag_id": 4,
+            "spice_id": 6,
+            "parent_codicehi_id": 3,
+            "status": "INCOMPLETE",
+        },
+        {
+            "id": 7,
+            "filename": "imap_l0_sci_codicehi_20230531_v01.cdf",
+            "instrument": "codicehi",
+            "version": 1,
+            "level": "l0",
+            "mode": "NULL",
+            "date": datetime.datetime(
+                2023, 6, 2, 5, 45, tzinfo=zoneinfo.ZoneInfo(key="America/Denver")
+            ),
+            "ingested": datetime.datetime(
+                2023, 6, 2, 5, 45, tzinfo=zoneinfo.ZoneInfo(key="America/Denver")
+            ),
+            "mag_id": 4,
+            "spice_id": 6,
+            "parent_codicehi_id": 3,
+            "status": "INCOMPLETE",
+        },
+    ]
+
+    res_true = all_dependency_present(result, dependencies_true)
+    res_false = all_dependency_present(result, dependencies_false)
+
+    assert res_true is True
+    assert res_false is False
+
+
+def test_query_dependents(setup_database):
+    # Test to query the database to make certain dependents are
+    # not already there.
+    conn = setup_database
+    cur = conn.cursor()
+
+    instrument_dependents = [
+        {"instrument": "CodiceHi", "level": "l3b"},
+        {"instrument": "CodiceHi", "level": "l3c"},
+    ]
+    process_dates = ["2023-05-31", "2023-06-01", "2023-06-02"]
+
+    # Dependents that have been ingested for this date range.
+    records = query_dependents(cur, 1, process_dates, instrument_dependents)
+
+    assert records[0]["filename"] == "imap_l3b_sci_codicehi_20230531_v01.cdf"
+
+    # Since there are 2 instruments x 3 dates and one record to remove = 5
+    output = remove_ingested(records, instrument_dependents, process_dates)
+
+    assert len(output) == 5
+
+
+def test_query_dependencies(setup_database):
```

Comment:
Easy testing is an argument _against_ consolidating all the database calls. Something to consider.